### PR TITLE
First pass at trim support

### PR DESCRIPTION
**Pull request by agocke**
This isn't ready for merging, just an example of the problems related to trimming

Protobuf problems at https://github.com/protocolbuffers/protobuf/pull/9183

### CODE REVIEW

File: testassets/LinkerTestsClient/LinkerTestsClient.csproj [hunk 1]
```diff
@@ -1,29 +1,18 @@
...
   </ItemGroup>
 
-  <Target Name="EnsureAllAssembliesAreLinked" BeforeTargets="PrepareForILLink">
-    <ItemGroup>
-      <ManagedAssemblyToLink>
-        <TrimMode>link</TrimMode>
-      </ManagedAssemblyToLink>
-      <TrimmerRootAssembly Include="$(TargetName)" />
-    </ItemGroup>
-  </Target>
```
Comment (JamesNK) [inline testassets/LinkerTestsClient/LinkerTestsClient.csproj:27]: No longer needed in .NET 6?

File: testassets/LinkerTestsClient/LinkerTestsClient.csproj [hunk 2]
```diff
@@ -1,29 +1,18 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>net5.0</TargetFramework>
+    <TargetFramework>net6.0</TargetFramework>
     <OutputType>Exe</OutputType>
-    
     <PublishTrimmed>true</PublishTrimmed>
-    <TrimMode>Link</TrimMode>
```
Comment (JamesNK) [inline testassets/LinkerTestsClient/LinkerTestsClient.csproj:8]: No longer needed in .NET 6?

**Comment by JamesNK** [comment]
I'm going to close this. Will use https://github.com/grpc/grpc-dotnet/pull/1465 for these changes.

**Comment by agocke** [comment]
Thanks!

File: testassets/LinkerTestsClient/LinkerTestsClient.csproj (hunk 2 above)
Comment (agocke) [reply testassets/LinkerTestsClient/LinkerTestsClient.csproj:8]: That's the default now

File: testassets/LinkerTestsClient/LinkerTestsClient.csproj (hunk 1 above)
Comment (agocke) [reply testassets/LinkerTestsClient/LinkerTestsClient.csproj:27]: `IsTrimmable` does the same thing on a per-assembly basis. I would opt-in in the assembly itself, to enable trimming for everyone

**Pull request closed without merging**